src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -475,9 +479,13 @@ public static void assertFilesEqual(final File f1, final File f2) {
             while ((len1 = s1.read(buf1)) != -1) {
                 final int len2 = s2.read(buf2);
                 if (len1 != len2) {
+                    s1.close();
```

Comment:
Good catch on these.  Instead of putting closes at all the exit points, I think it would make more sense to lift the creation of the FileInputStreams into a try-with-resources block and then they'll be guaranteed to be closed no matter where the function exits.